patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1010,73 +1010,348 @@
  			if (diff == 1)
  				baseColor = new Microsoft.Xna.Framework.Color((byte)((float)(int)mcColor.R * num2), (byte)((float)(int)mcColor.G * num2), (byte)((float)(int)mcColor.B * num2), mouseTextColor);
  
-@@ -14844,8 +_,9 @@
+@@ -14837,107 +_,114 @@
+ 
+ 			int num2 = 30;
+ 			int numLines = 1;
+-			string[] array = new string[num2];
++			string[] tooltipLines = new string[num2];
+ 			bool[] array2 = new bool[num2];
+ 			bool[] array3 = new bool[num2];
+ 			for (int i = 0; i < num2; i++) {
  				array2[i] = false;
  				array3[i] = false;
  			}
 +			string[] tooltipNames = new string[num2];
  
 -			MouseText_DrawItemTooltip_GetLinesInfo(HoverItem, ref yoyoLogo, ref researchLine, knockBack, ref numLines, array, array2, array3);
-+			MouseText_DrawItemTooltip_GetLinesInfo(HoverItem, ref yoyoLogo, ref researchLine, knockBack, ref numLines, array, array2, array3, tooltipNames);
++			MouseText_DrawItemTooltip_GetLinesInfo(HoverItem, ref yoyoLogo, ref researchLine, knockBack, ref numLines, tooltipLines, array2, array3, tooltipNames);
  			float num3 = (float)(int)mouseTextColor / 255f;
- 			float num4 = num3;
+-			float num4 = num3;
++			float colorThing = num3;
  			int a = mouseTextColor;
-@@ -14853,6 +_,7 @@
+ 			if (npcShop > 0 && hoverItem.value >= 0 && (hoverItem.type < 71 || hoverItem.type > 74)) {
  				LocalPlayer.GetItemExpectedPrice(hoverItem, out int calcForSelling, out int calcForBuying);
  				int num5 = (hoverItem.isAShopItem || hoverItem.buyOnce) ? calcForBuying : calcForSelling;
  				if (hoverItem.shopSpecialCurrency != -1) {
 +					tooltipNames[numLines] = "SpecialPrice";
- 					CustomCurrencyManager.GetPriceText(hoverItem.shopSpecialCurrency, array, ref numLines, num5);
- 					color = new Microsoft.Xna.Framework.Color((byte)(255f * num4), (byte)(255f * num4), (byte)(255f * num4), a);
- 				}
-@@ -14912,7 +_,8 @@
- 						array[numLines] = Lang.tip[49].Value + " " + text;
+-					CustomCurrencyManager.GetPriceText(hoverItem.shopSpecialCurrency, array, ref numLines, num5);
++					CustomCurrencyManager.GetPriceText(hoverItem.shopSpecialCurrency, tooltipLines, ref numLines, num5);
+-					color = new Microsoft.Xna.Framework.Color((byte)(255f * num4), (byte)(255f * num4), (byte)(255f * num4), a);
++					color = new Microsoft.Xna.Framework.Color((byte)(255f * colorThing), (byte)(255f * colorThing), (byte)(255f * colorThing), a);
+ 				}
+ 				else if (num5 > 0) {
+ 					string text = "";
+-					int num6 = 0;
++					int costPlatinum = 0;
```

Comment:
You're gonna have to undo these because they make the patch necessarily big
We generally want to keep patches small, and renaming num variables is a bit irrelevant since they pretty much can change every update, this counts for literally all renames you did (it's also impossible for me to find now what exactly you did)

Reply:
it has been done now, should make what I actually did much more apparent
does make the method a bit less readable but w/e